src/Sarif/Writers/BaseLogger.cs

```diff
@@ -15,36 +15,35 @@ public abstract class BaseLogger
         protected BaseLogger(IEnumerable<FailureLevel> failureLevels,
             IEnumerable<ResultKind> resultKinds)
         {
-            if (failureLevels == null || failureLevels.Count() == 0)
-            {
-                _failureLevels = new List<FailureLevel> { FailureLevel.Error, FailureLevel.Warning };
-            }
-            else
-            {
-                _failureLevels = failureLevels.ToList();
-            }
-
-            if (resultKinds == null || resultKinds.Count() == 0)
-            {
-                _resultKinds = new List<ResultKind> { ResultKind.Fail };
-            }
-            else
-            {
-                _resultKinds = resultKinds.ToList();
-            }
+            _failureLevels = failureLevels?.ToList() ?? new List<FailureLevel>();
+            _resultKinds = resultKinds?.ToList() ?? new List<ResultKind>();
 
             ValidateParameters();
         }
 
         private void ValidateParameters()
         {
-            //  If we got here, both resultKinds and failurelevels can neither be null nor empty.
-            //  If resultKinds does not include "fail" then failureLevels MUST be none/zero
-            if (!_resultKinds.Contains(ResultKind.Fail))
+            if (_resultKinds.Count==0 || (_resultKinds.Count==1 && _resultKinds[0] == ResultKind.None))
```

Comment:
>== [](start = 34, length = 2)

use ctrl + k + d to auto ident

Reply:
Executed buildandtest.cmd instead

---
In reply to: [563044550](https://github.com/microsoft/sarif-sdk/pull/2262#discussion_r563044550) [](ancestors = 563044550)